src/components/MultiHopTrade/components/TradeInput/components/Claim/hooks/useArbitrumClaimTx.tsx

```diff
@@ -0,0 +1,149 @@
+    return { accountId: destinationAccountId }
+  }, [destinationAccountId])
+
+  const bip44Params = useAppSelector(state => selectBIP44ParamsByAccountId(state, accountIdFilter))
+
+  const executeTransactionDataResult = useQuery({
+    queryKey: ['executeTransactionData', { txid: claim.tx.txid }],
+    queryFn: async () => {
+      const { event, message } = claim
+
+      const proof = await message.getOutboxProof(l2Provider)
+
+      if (!('position' in event)) return
+
+      // nitro transaction
+      return encodeFunctionData({
+        abi: outboxAbi,
+        functionName: 'executeTransaction',
+        args: [
+          proof,
+          event.position,
+          event.caller,
+          event.destination,
+          event.arbBlockNum,
+          event.ethBlockNum,
+          event.timestamp,
+          event.callvalue,
+          event.data,
+        ],
+      })
+    },
+  })
+
+  const evmFeesResult = useEvmFees({
+    accountNumber: bip44Params?.accountNumber,
+    chainId: claim.destinationChainId,
+    data: executeTransactionDataResult.data,
+    refetchInterval: 15_000,
+    to: getAddress(ARBITRUM_OUTBOX),
+    value: '0',
+  })
+
+  const claimMutation = useMutation({
+    mutationKey: ['claim', { txid: claim.tx.txid }],
+    mutationFn: async () => {
+      if (!wallet) return
+      if (!bip44Params) return
+      if (!executeTransactionDataResult.data) return
+
+      const adapter = assertGetEvmChainAdapter(claim.destinationChainId)
+
+      const buildCustomTxInput = await createBuildCustomTxInput({
+        accountNumber: bip44Params.accountNumber,
+        adapter,
+        data: executeTransactionDataResult.data,
+        to: ARBITRUM_OUTBOX,
+        value: '0',
+        wallet,
+      })
+
+      const txHash = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: CONTRACT_INTERACTION,
+      })
+
+      return txHash
+    },
+    onMutate() {
+      setClaimTxStatus(TxStatus.Pending)
+    },
+    onSuccess(txHash) {
+      if (!txHash) {
+        setClaimTxStatus(TxStatus.Failed)
+        return
+      }
+
+      setClaimTxHash(txHash)
+
+      const checkStatus = async () => {
+        const publicClient = assertGetViemClient(claim.destinationChainId)
+        const { status } = await publicClient.waitForTransactionReceipt({ hash: txHash as Hash })
+
+        switch (status) {
+          case 'success': {
+            queryClient.setQueryData(['claimStatus', { txid: claim.tx.txid }], () => null)
+            return setClaimTxStatus(TxStatus.Confirmed)
+          }
+          case 'reverted':
+          default:
+            return setClaimTxStatus(TxStatus.Failed)
+        }
+      }
+
+      checkStatus()
+    },
+    onError() {
+      setClaimTxStatus(TxStatus.Failed)
+    },
+    onSettled() {
+      queryClient.invalidateQueries({
+        queryKey: ['claimStatus', { txid: claim.tx.txid }],
+        refetchType: 'all',
+      })
+    },
+  })
```

Comment:
q: do we still need to `invalidateQueries` on `handleGoBack()` in `<ClaimStatus />`, givem this does the same `onSettled()`?

Reply:
Probably not necessary, was just a paranoia addition if they selected back before the mutation settled. Does the mutation finish even though the component unmounts, or does it cancel the mutation subscription and subsequent updates out of curiosity?

Reply:
That's a very fair point. Gut feel was `useMutation()` is inherently a little diff. from `useQuery()` (query observer vs. mutation promise), and indeed seems to be the case https://github.com/TanStack/query/discussions/5133#discussioncomment-5323478

Note howevever, that this is only valid when defining said callbacks in `useMutation()`, not when doing so in `mutate()` / `mutateAsync()`:
https://tanstack.com/query/latest/docs/framework/react/guides/mutations#mutation-side-effects

> You might find that you want to trigger additional callbacks beyond the ones defined on useMutation when calling mutate. This can be used to trigger component-specific side effects. To do that, you can provide any of the same callback options to the mutate function after your mutation variable. Supported options include: onSuccess, onError and onSettled. Please keep in mind that those additional callbacks won't run if your component unmounts before the mutation finishes.

^ this assumes the callbacks are defined in `useMutation()` which is our case here so we should be g2g!

Reply:
https://github.com/shapeshift/web/pull/7563/commits/137c23ece063b023eb146df6f8c2d2c49ac78854